DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+            if (newAuction.status == Mode.Owned && now < newAuction.renewalDate) {
+                throw; 
+            } else if (newAuction.status == Mode.Auction && now < newAuction.auctionDate) {
+                throw;
+            } else if (newAuction.status == Mode.Owned) {
+                Deed deedContract = Deed(newAuction.deed);
+                deedContract.closeDeed(1);
+            }
+            newAuction.auctionDate = now + auctionLength;
+            newAuction.renewalDate = now + renewalPeriod;
+            newAuction.status = Mode.Auction;
+
+        }
+    }
+    
+    function shaBid(bytes32 name, address owner, uint value, bytes32 salt) constant returns (bytes32 sealedBid) {
+        return sha3(name, owner, value, salt);
+    }
+    
+    /*
+    ## One week auction for the desired hash
+
+    Bids are sent by sending a message to the main contract with a hash and an amount. The hash 
+    contains information about the bid, including the bidded hash, the bid amount, and a random 
+    salt. Bids are not tied to any one auction until they are revealed. The value of the bid 
+    itself can be maskeraded by changing the required period or sending more than what you are 
+    bidding for. This is followed by a 24h reveal period. Bids revealed after this period will 
+    be burned and the ether unrecoverable. Since this is an auction, it is expected that most 
+    public names, like known domains and common dictionary words, will have multiple bidders pushing the price up. 
+    */ 
+    function newBid(bytes32 sealedBid) {
+        if (sealedBids[sealedBid] > 0 ) throw;
+        // creates a new Name contract with the owner
+        address newBid = new Deed(msg.sender, sealedBid, this);
+        sealedBids[sealedBid] = newBid;
+        if (!newBid.send(msg.value)) throw;
+    } 
+    
+    
+    /*
+    ## Winning bids are deposited
+
+    The highest bid gets control of the name, by depositing the equivalent amount of the second 
+    highest bid. The funds will be held on new contract, controlled only by the owner which 
+    contains basic information like when the name was first registered, when it needs to be 
+    renewed, and the price paid for it. **The price paid is also saved on a moving average amount, 
+    calculated as ```averagePrice = averagePrice * 0.999 + newPrice * 0.001```**. The 
+    averagePrice at the moment of purchase is also registered on the contract.
+    */ 
+    function revealBid(bytes32 _hashedName, address _owner, uint _value, bytes32 _salt) noEther  {
+        bytes32 seal = shaBid(_hashedName, _owner, _value, _salt);
+        address bidAddress = sealedBids[seal];
+        if (bidAddress == 0 ) throw;
+        Deed bid = Deed(bidAddress);
+        bid.unsealBid(_hashedName, _owner, _value, _salt);
+        entry name = hashedNames[bid.hashedName()];
+        
+        if (bid.creationDate() > name.auctionDate - revealPeriod
```

Comment:
Yes. Otherwise you can make a bid and choose not to reveal during the reveal period, after some others have revealed. This would change the auction from a blind vickrey into a completely different auction. As an example of what someone could do, they could bid multiple values for any auction and then blackmailing the current winner into paying them NOT to reveal a bid that would be low enough not to win, but high enough to increase the ask price of the name.